packages/client/components/NewGitLabIssueMenu.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import {MenuProps} from '~/hooks/useMenu'
+import RepoIntegrationGitLabMenuItem from './RepoIntegrationGitLabMenuItem'
+import useSearchFilter from '~/hooks/useSearchFilter'
+import {EmptyDropdownMenuItemLabel} from './EmptyDropdownMenuItemLabel'
+import Menu from './Menu'
+import {SearchMenuItem} from './SearchMenuItem'
+
+interface Props {
+  handleSelectFullPath: (key: string) => void
+  menuProps: MenuProps
+  gitlabProjects: {id: string; fullPath: string}[]
+}
+
+const getValue = (item: {fullPath?: string}) => {
+  return item.fullPath || 'Unknown Project'
+}
+
+const NewGitLabIssueMenu = (props: Props) => {
+  const {handleSelectFullPath, menuProps, gitlabProjects} = props
+
+  const {query, filteredItems: filteredProjects, onQueryChange} = useSearchFilter(
+    gitlabProjects,
+    getValue
+  )
+
+  const onClick = (fullPath: string) => {
+    handleSelectFullPath(fullPath)
+  }
+
+  return (
+    <Menu ariaLabel='Select GitLab project' keepParentFocus {...menuProps}>
```

Comment:
-1 this menu is waaaay too wide! the items are showing ellipsis, but the width is as big as can be.
the input field should dictate the width

![Screen Shot 2022-04-06 at 3 08 55 PM](https://user-images.githubusercontent.com/5514175/162081023-34d01767-08b9-4e04-b476-cefd72294613.png)